process/block/baseProcess.go

```diff
@@ -2396,42 +2352,37 @@ func (bp *baseProcessor) waitAllMissingProofs(waitTime time.Duration) error {
 	}
 }
 
-func (bp *baseProcessor) checkReceivedHeaderIfAttestingIsNeeded(headerHandler data.HeaderHandler) {
-	if !common.ShouldBlockHavePrevProof(headerHandler, bp.enableEpochsHandler, common.EquivalentMessagesFlag) {
-		return
-	}
-
-	bp.mutRequestedAttestingNoncesMap.RLock()
-	isWaitingForProofs := len(bp.requestedAttestingNoncesMap) > 0
-	bp.mutRequestedAttestingNoncesMap.RUnlock()
+func (bp *baseProcessor) checkReceivedProofIfAttestingIsNeeded(proof data.HeaderProofHandler) {
+	bp.mutRequestedProofsMap.RLock()
+	isWaitingForProofs := len(bp.requestedProofsMap) > 0
+	bp.mutRequestedProofsMap.RUnlock()
 	if !isWaitingForProofs {
 		return
 	}
 
-	allProofsReceived := bp.checkReceivedHeaderAndUpdateMissingAttesting(headerHandler)
+	allProofsReceived := bp.checkReceivedProofAndUpdateMissing(proof)
 	if allProofsReceived {
 		bp.allProofsReceived <- true
 	}
 }
 
-func (bp *baseProcessor) checkReceivedHeaderAndUpdateMissingAttesting(headerHandler data.HeaderHandler) bool {
-	bp.mutRequestedAttestingNoncesMap.Lock()
-	defer bp.mutRequestedAttestingNoncesMap.Unlock()
+func (bp *baseProcessor) checkReceivedProofAndUpdateMissing(proof data.HeaderProofHandler) bool {
+	bp.mutRequestedProofsMap.Lock()
+	defer bp.mutRequestedProofsMap.Unlock()
 
-	receivedShard := headerHandler.GetShardID()
-	prevHash := headerHandler.GetPrevHash()
-	_, isHeaderWithoutProof := bp.requestedAttestingNoncesMap[string(prevHash)]
-	if !isHeaderWithoutProof {
-		log.Debug("received header does not have previous hash any of the requested ones")
-		return len(bp.requestedAttestingNoncesMap) == 0
+	_, isMissingProof := bp.requestedProofsMap[string(proof.GetHeaderHash())]
+	if !isMissingProof {
+		log.Trace("received proof but not from the requested ones")
+		return len(bp.requestedProofsMap) == 0
 	}
 
-	if !bp.proofsPool.HasProof(receivedShard, prevHash) {
-		log.Debug("received next header but proof is still missing", "hash", hex.EncodeToString(prevHash))
-		return len(bp.requestedAttestingNoncesMap) == 0
+	if !bp.proofsPool.HasProof(proof.GetHeaderShardId(), proof.GetHeaderHash()) {
+		// this should never happen
+		log.Debug("received proof but proof is still missing", "hash", hex.EncodeToString(proof.GetHeaderHash()))
```

Comment:
maybe also change to log.error if this should never happen.

Reply:
changed